homeassistant/helpers/entity.py

```diff
@@ -889,7 +903,7 @@ def _async_write_ha_state(self) -> None:
                 entity_registry = er.async_get(self.hass)
                 self.registry_entry = entity_registry.async_update_entity(
                     self.entity_id,
-                    capabilities=capabilities,
+                    capabilities=state.capability_attributes,
```

Comment:
This is the second time we access the attribute. Then we should first use a local variable.